### PR TITLE
fix(cdk): remove global styles patching from auto-focus directive

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
Close #

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/demo/src/modules/components/dialog/examples/7/search-example/search-dialog-example.component.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {injectContext} from '@taiga-ui/polymorpheus';
     templateUrl: './search-dialog-example.template.html',
     styleUrls: ['./search-dialog-example.component.less'],
     changeDetection,
+    providers: [tuiAutoFocusOptionsProvider({preventScroll: true})],
```
Comment (splincode) [inline projects/demo/src/modules/components/dialog/examples/7/search-example/search-dialog-example.component.ts:14]: image

add info please

File: projects/cdk/directives/auto-focus/handlers/default.handler.ts [hunk 2]
```diff
@@ -29,7 +29,9 @@ export class TuiDefaultAutofocusHandler extends AbstractTuiAutofocusHandler {
                     skipWhile(Boolean),
                     take(1),
                 ),
-            ).subscribe(() => this.element.focus({preventScroll: true}));
+            ).subscribe(() =>
+                this.element.focus({preventScroll: this.options.preventScroll}),
+            );
         } else {
             this.element.focus({preventScroll: true});
```
Comment (splincode) [inline projects/cdk/directives/auto-focus/handlers/default.handler.ts:36]: Should we add `{preventScroll: this.options.preventScroll}` for `this.element.focus({preventScroll: true});`?

Comment (vladimirpotekhin) [reply projects/cdk/directives/auto-focus/handlers/default.handler.ts:36]: no, that would be a breaking change for the buttons

File: projects/core/styles/mixins/mixins.less [hunk 3]
```diff
@@ -129,3 +129,20 @@
     overflow: hidden;
     padding: 0;
 }
+
+// prevent scrolling to focused element on IOS
+.tui-prevent-ios-scroll() {
```
Comment (mdlufy) [inline projects/core/styles/mixins/mixins.less:134]: Why we need prevent scroll?

Comment (vladimirpotekhin) [reply projects/core/styles/mixins/mixins.less:134]: This is for fixed textfields like searchbar. IOS Safari tries to scroll to the textfield and the interface shakes

**Pull request merged**